src/main/java/Terry.java

```diff
@@ -0,0 +1,139 @@
+        ui.printListSize(list);
+    }
+
+    private static void handleMarkCase(String[] words, UserInputList list, Ui ui) throws TerryException {
+        if (words.length < COMMAND_AND_DETAILS_LIMIT) { // can also use this limit since need mark followed by index as well!
+            throw new TerryException("YOU DIDN'T PUT THE NUMBER THERE");
+        } else {
+            try {
+                int index = Integer.parseInt(words[1]);
+                if (list.getOneTask(index) != null) {
+                    list.mark(index);
+                    System.out.println("aww yea the task is marked LOOK");
+                    ui.printIndex(list, index);
+                } else {
+                    throw new TerryException("WE HAVEN'T REACHED SO MANY TASKS YET");
+                }
+            } catch (NumberFormatException e) {
+                throw new TerryException("YOU DIDN'T PUT THE NUMBER THERE");
+            }
+        }
+    }
+
+    private static void addItemToList(UserInputList list, String line, Ui ui, Storage storage) throws TerryException {
+        String[] words = line.split(" ", COMMAND_AND_DETAILS_LIMIT);
+        if (words.length < COMMAND_AND_DETAILS_LIMIT) {
+            System.out.println("Invalid format! Please specify the task details.");
+            throw new TerryException("Invalid format! Please specify the task details.");
+
+        }
+
+        String taskType = words[0]; // First part is the command
+        String details = words[1];
+
+        Task newTask = null; // Task object to be added
+
+        switch (taskType) {
+        case "todo":
+            newTask = new ToDo(details);
+            break;
+        case "deadline":
+            String[] deadlineParts = details.split(" /by ", 2);
+            if (deadlineParts.length < 2) {
+                throw new TerryException("Invalid format for deadline! Use: deadline [description] /by [date]");
+            }
+            newTask = new Deadline(deadlineParts[0], deadlineParts[1]);
+            break;
+        case "event":
+            String[] eventParts = details.split(" /from ", 2);
+            if (eventParts.length < 2 || !eventParts[1].contains(" /to ")) {
+                throw new TerryException("Invalid format for event! Use: event [description] /from [start] /to [end]");
+            }
+            String[] times = eventParts[1].split(" /to ", 2);
+            newTask = new Event(eventParts[0], times[0], times[1]);
+            break;
+        default:
+            throw new TerryException("Unknown task type! Use 'todo', 'deadline', or 'event'.");
+        }
+
+        list.add(newTask); // Add the task to the list
+
+        ui.printTasks(list, newTask);
+        // java automatically calls the toString function when printing an object type,
+        //so printing newTask will call toString, invoking the toString functions in the
+        //respective subclasses since they have @override inside
+//        list.saveTasksToFile();
```

Comment:
Ideally comments should not read like notes to the developer. This does not read like an explanation for the code but more like how certain things in java work (like `toString()`). Additionally, do not comment out dead code. Remove it if it is not needed.